test/integration/client/exec_test.go

```diff
@@ -537,6 +540,118 @@ func TestExecPluginViaInformer(t *testing.T) {
 	}
 }
 
+type execPlugin struct {
+	t          *testing.T
+	outputFile *os.File
+}
+
+func newExecPlugin(t *testing.T) *execPlugin {
+	t.Helper()
+	outputFile, err := ioutil.TempFile("", "kubernetes-client-exec-test-plugin-output-file-*")
+	if err != nil {
+		t.Fatal(err)
+	}
+	return &execPlugin{t: t, outputFile: outputFile}
+}
+
+func (e *execPlugin) config() *clientcmdapi.ExecConfig {
+	return &clientcmdapi.ExecConfig{
+		Command: "testdata/exec-plugin.sh",
+		// TODO(ankeesler): move to v1 once exec plugins go GA.
+		APIVersion: "client.authentication.k8s.io/v1beta1",
+		Env: []clientcmdapi.ExecEnvVar{
+			{
+				Name:  outputFileEnvVar,
+				Value: e.outputFile.Name(),
+			},
+		},
+	}
+}
+
+func (e *execPlugin) rotateToken(newToken string, lifetime time.Duration) {
+	e.t.Helper()
+
+	expirationTimestamp := metav1.NewTime(time.Now().Add(lifetime)).Format(time.RFC3339Nano)
+	newOutput := fmt.Sprintf(`{
+		"kind": "ExecCredential",
+		"apiVersion": "client.authentication.k8s.io/v1beta1",
+		"status": {
+			"expirationTimestamp": %q,
+			"token": %q
+		}
+	}`, expirationTimestamp, newToken)
+	if err := os.WriteFile(e.outputFile.Name(), []byte(newOutput), 0644); err != nil {
+		e.t.Fatal(err)
+	}
+
+	// Wait for old token to expire.
+	time.Sleep(lifetime)
```

Comment:
This line pauses the test so that the exec authenticator will need to be triggered again to pick up the new token. If we didn't have this, a line like this might fail because the informer recreates its watch within the lifetime of the `clientAuthorizedToken`: https://github.com/kubernetes/kubernetes/blob/3df78461ad2cdf161d09c681372b3b99412cb435/test/integration/client/exec_test.go#L642.

I suppose everywhere else we could get away without this pause since the informer would eventually pick up the new token over the course of our timeouts (e.g., https://github.com/kubernetes/kubernetes/blob/3df78461ad2cdf161d09c681372b3b99412cb435/test/integration/client/exec_test.go#L631).

I wrote this a different way - do you think this is better? This would get rid of 15s of unnecessary waiting (new test runs in `125.988s`) and potentially some confusing test code: 180e7c4aa86.